### PR TITLE
The migration of the Jenkins pipeline to the Buildkite pipeline

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

The migration of the Jenkins pipeline to the Buildkite pipeline. 

## How does this PR solve the problem?

Preparing the Buldkite pipeline instead Jenkins pipeline

## How to test this PR locally

It doesn't be checked locally, that's why it's a partial "step by step" migration

## Checklist

The Buildkite pipeline will be triggered by this PR and we will see the result

## Related issues

-

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -37,12 +37,12 @@ steps:
...
           image: "${GO_WINDOWS_AGENT_IMAGE}"
         artifact_paths:
-          - build/test-results/test-windows-*.xml
+          - build/test-results/*.xml
 
       - label: ":junit: Junit annotate"
         plugins:
           - junit-annotate#v2.4.1:
-              artifacts: "build/test-results/test-*.xml"
+              artifacts: "bbuild/test-results/*.xml"
```
Comment (jlind23) [inline .buildkite/pipeline.yml:45]: was the double "b" added on purpose here?

**Review by jlind23 — APPROVED** [review]

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (sharbuz) [reply .buildkite/pipeline.yml]: no, by mistake.
I already got the error in the last stage ))
fixed.

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+
+env:
+  SETUP_GVM_VERSION: 'v0.5.0' # https://github.com/andrewkroh/gvm/issues/44#issuecomment-1013231151
+  SETUP_MAGE_VERSION: "latest"
+  GH_CLI_VERSION: "2.29.0"
```
Comment (mrodm) [inline .buildkite/pipeline.yml:6]: Do we use gh cli here ? IIRC it is not, and I could be removed.

**Review by mrodm — COMMENTED** [review]
we could add as part of this release a change in the `catalog-info.yaml` to add the needed environment variable to add comments with the status of the build:

```diff
diff --git catalog-info.yaml catalog-info.yaml
index ba2b11a..58dcb88 100644
--- catalog-info.yaml
+++ catalog-info.yaml
@@ -56,6 +56,8 @@ spec:
       cancel_intermediate_builds_branch_filter: '!main'
       skip_intermediate_builds: true
       skip_intermediate_builds_branch_filter: '!main'
+      env:
+        ELASTIC_PR_COMMENTS_ENABLED: 'true'
       teams:
         ecosystem:
           access_level: MANAGE_BUILD_AND_READ
```

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (sharbuz) [reply .buildkite/pipeline.yml]: removed

**Comment by sharbuz** [comment]
Regarding the comment:
https://github.com/elastic/package-spec/pull/552#pullrequestreview-1504763888
If this PR is fine, probably it's better to create one more PR with changes only in the catalog-info.yml: 
1. your suggestion 
2. add Slack notifications about fails.
@mrodm WDYT?

**Comment by mrodm** [comment]
> Regarding the comment: [#552 (review)](https://github.com/elastic/package-spec/pull/552#pullrequestreview-1504763888) If this PR is fine, probably it's better to create one more PR with changes only in the catalog-info.yml:
> 
> 1. your suggestion
> 2. add Slack notifications about fails.
>    @mrodm WDYT?
@sharbuz It can be added in this PR the update for catalog-info.yaml.

About slack notifications when it fails, currently Jenkins is not configured to do so. Therefore I would keep that way and not add those notifications

**Review by mrodm — APPROVED** [review]
LGTM

**Pull request merged**